rlog/2024-06-24-gsub-largemsg-improvements.mdx

```diff
@@ -0,0 +1,341 @@
+---
+title: 'Large Message Handling in GossipSub: Potential Improvements'
+date: 2024-06-24 15:00:00
+authors: farooq
+published: false
+slug: GossipSub Improvements
+categories: research
+
+
+toc_min_heading_level: 2
+toc_max_heading_level: 5
+---
+
+Large Message Handling in GossipSub: Potential Improvements 
+
+<!--truncate-->
+
+## Motivation:
+The challenge of large message transmissions in gossipsub leads to longer than expected network-wide message dissemination times (and relatively higher fluctuations). 
+It is particularly relevant for applications like Waku and Ethereum [], 
+that require on-time network-wide dissemination of large messages. 
+
+This matter has been extensively discussed in the libp2p community [], 
+and numerous improvements have been suggested (or even incorporated) to the gossipsub protocol to enable efficient large-message propagation.
+
+## Problem Realization:
+
+Sending a message to N peers involves approximately $\lceil \log_D(N) \rceil$ transmission rounds, 
+with around $(D-1)^{X-1} \times D$ transmissions in each round, 
+where $X, D, N$ represent the round number, mesh size, and network size. 
+
+Transmitting to a higher number of peers (floodpublish) can theoretically reduce latency by increasing the transmissions in each round to $(D-1)^{X-1} \times (F+D)$, 
+where $F$ represents the number of peers included in floodpublish. 
+
+This arrangement works fine for relatively small/moderate message sizes. 
+However, as message sizes increase, a significant rise and fluctuations in network-wide message dissemination time are seen. 
+Interestingly, a higher $D$ or $F$ also degrade performance in this situation. 
+
+Several aspects contribute to this behavior: 
+
+1. Ideally, a message transmission to a single peer concludes in $\tau_1 = \frac {L}{R}+P$ (ignoring any message processing time), 
+where $L, R, P$ represent message size, data rate, and link latency. 
+Therefore, the time required for sending a message on a 100Mbps link with 100ms latency 
+jumps from $\tau_1^{10KB} = 100.8ms$ for a 10KB message to $\tau_1^{1MB} = 180ms$ for a 1MB message. 
+For $D$ peers, the transmission time multiplies to $\tau_D^{1MB} = (80 \times D) + 100ms$, 
+triggering additional queuing delays (proportional to the transmission queue size) during each transmission round.  
+
+2. In practice, $\tau_1^{1MB}$ sometimes rises to several hundred milliseconds, 
+further exaggerating the abovementioned queuing delays.
+This rise is because TCP congestion avoidance limits the maximum in-flight bytes to approximately ${C_{wnd} \times MSS}$ in a single RTT, 

```

Comment:
RTT hasn't been used yet, so might be good to write it out in the first occurrence.